lib/cn/cn_tree.c

```diff
@@ -100,22 +102,25 @@ cn_node_alloc(struct cn_tree *tree, uint64_t nodeid)
 
     INIT_LIST_HEAD(&tn->tn_link);
     INIT_LIST_HEAD(&tn->tn_kvset_list);
-    INIT_LIST_HEAD(&tn->tn_ss_list);
-
-    mutex_init(&tn->tn_ss_lock);
 
     atomic_init(&tn->tn_compacting, 0);
     atomic_init(&tn->tn_busycnt, 0);
-    atomic_init(&tn->tn_split_cnt, 0);
-    tn->tn_split_dly = 0;
 
     tn->tn_tree = tree;
     tn->tn_isroot = (nodeid == 0);
     tn->tn_nodeid = nodeid;
 
     tn->tn_split_size = (size_t)tree->rp->cn_split_size << 30;
-    cv_init(&tn->tn_spill_cv);
+
     mutex_init(&tn->tn_spill_mtx);
+    cv_init(&tn->tn_spill_cv);
+
+    mutex_init(&tn->tn_ss_lock);
+    INIT_LIST_HEAD(&tn->tn_ss_list);
+    tn->tn_ss_spilling = 0;
+    tn->tn_ss_splitting = false;
+    tn->tn_ss_visits = 0;
+    cv_init(&tn->tn_ss_cv);
```

Comment:
zalloc'd right?  No need to explicitly set members to 0 or false.

Reply:
True.  Sometimes I find it useful to see where stuff if initialized, but I can remove them if it bugs you :-)

Reply:
Actually, those are atomics, would prefer to initialize them.

Reply:
w/ atomic_set ?